spec/facter/resolvers/cloud_spec.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+describe Facter::Resolvers::Cloud do
+  subject(:cloud_resolver) { Facter::Resolvers::Cloud }
+
+  let(:log_spy) { instance_spy(Facter::Log) }
+
+  before do
+    cloud_resolver.instance_variable_set(:@log, log_spy)
+    allow(File).to receive(:directory?)
+    allow(Dir).to receive(:entries)
+  end
+
+  after do
+    cloud_resolver.invalidate_cache
+  end
+
+  context 'when lease files are not found' do
+    before do
+      allow(File).to receive(:directory?).with('/var/lib/dhcp').and_return(true)
+      allow(Dir).to receive(:entries).with('/var/lib/dhcp').and_return('.')
+    end
+
+    it 'returns nil' do
+      expect(cloud_resolver.resolve(:cloud_provider)).to be_nil
+    end
+  end
+
+  context 'when lease file is found and contains option 245' do
+    let(:content) { load_fixture('dhclient_rhel_lease_8').read }
+
+    before do
+      allow(File).to receive(:directory?).with('/var/lib/dhcp').and_return(true)
+      allow(Facter::Util::FileHelper)
+        .to receive(:safe_readlines)
+        .with('/var/lib/dhcp/dhclient.rhel.lease.8')
+        .and_return(content)
+      allow(Dir).to receive(:entries).with('/var/lib/dhcp').and_return(['.', 'dhclient.rhel.lease.8'])
+    end
+
+    it 'returns azure' do
+      expect(cloud_resolver.resolve(:cloud_provider)).to eq('azure')
+    end
+  end
+
+  context 'when lease file is found without option 245' do
```

Comment:
Could you add a test when the lease file is found but it can't be read?

Reply:
Yes, updated.